src/mono/wasm/runtime/run-outer.ts

```diff
@@ -243,7 +243,10 @@ class HostBuilder implements DotnetHostBuilder {
     withRuntimeOptions(runtimeOptions: string[]): DotnetHostBuilder {
         try {
             mono_assert(runtimeOptions && Array.isArray(runtimeOptions), "must be array of strings");
-            Object.assign(this.moduleConfig, { runtimeOptions });
+            const configInternal: MonoConfigInternal = {
+                runtimeOptions: runtimeOptions
+            };
+            Object.assign(this.moduleConfig.config!, configInternal);
```

Comment:
`configInternal.runtimeOptions= [...(this.moduleConfig.config!.runtimeOptions|| []), ...(configInternal.assets || [])];`